Aprendizado por Reforço Clássico/Monte Carlo/Monte Carlo Every-Visit/Monte Carlo Every-Visit.ipynb

```diff
@@ -0,0 +1,254 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# 🏔 Monte Carlo Every Visit\n",
+    "\n",
+    "**Monte Carlo Every-Visit** é um algoritmo de controle por Monte Carlo, ou seja, ele estima nossa função de valor *q(s, a)* a partir dos retornos médios de cada par estado-ação, e toma ações no ambiente com base nessas estimativas. \n",
+    "\n",
+    "Entretanto, esse algoritmo difere de outros métodos de Monte Carlo por utilizar todos os retornos de um par estado-ação durante um episódio. Isso significa que, quando o nosso agente visita um estado repetido e toma uma mesma ação, o cálculo da função de valor levará em conta o retorno de todas as vezes que essa ação foi tomada."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Política ε-greedy\n",
+    "\n",
+    "Para garantir que os métodos de Monte Carlo convirjam para a função de valor real, é necessário seguir uma política que explore todas as ações de todos os estados. Entretanto, também é interessante que o agente tente conseguir cada vez mais recompensas, para maximizar sua perfomance.\n",
+    "\n",
+    "Assim foi desenvolvida a política *ε-greedy*, que escolhe a próxima ação com base em um parâmetro *ε*, normalmente pequeno. A cada decisão, a política tem uma probabilidade *ε* de escolher uma ação aleatória, aumentando a exploração, e uma probabilidade *1 - ε* de escolher a ação associada ao maior *Q*. Dessa forma, ela estabelece um equilíbrio entre a exploração de ações e a explotação de recompensas. Essa política é dada por:\n",
+    "\n",
+    "$$\\pi(a|S_t) \\leftarrow \\begin{cases} 1 - \\varepsilon + \\varepsilon/\\left|\\mathcal{A}(S_t)\\right|, & \\mbox{se } a = \\underset{a}{\\mathrm{argmax}} \\, Q(S_t,a) \\\\ \\varepsilon/\\left|\\mathcal{A}(S_t)\\right|, & \\mbox{se } a \\neq \\underset{a}{\\mathrm{argmax}} \\, Q(S_t,a) \\end{cases}$$\n",
+    "\n",
+    "$$\\left|\\mathcal{A}(S_t)\\right| \\rightarrow \\textrm{quantidade de ações possíveis}$$"
```

Comment:
```suggestion
    "$$\\left|\\mathcal{A}(S_t)\\right| \\rightarrow \\textrm{quantidade\\ de\\ ações\\ possíveis}$$"
```

Reply:
Não sei bem o que isso faz?

Reply:
No meu ele já parece certo

![image](https://user-images.githubusercontent.com/48636340/92183476-ecf60180-ee24-11ea-880d-7a26a462741c.png)

Reply:
Isso é sobre o .ipynb, que fica assim (sim, tá diferente do md):
![image](https://user-images.githubusercontent.com/12588556/92184124-d7da9c00-ee3f-11ea-94f6-351f42eea602.png)
- No markdown, você usou `\textrm{...}`. Isso coloca o latex em modo de texto, que considera espaços.
- No notebook, você usou `\mathrm{...}`. Isso mantém o latex em modo de matemática (mas muda a formatação pra não ser itálico). Como no modo de matemática o espaçamento é diferente, os espaços são ignorados e o texto fica zuado. Dá pra consertar isso de duas formas: (i) forçando o latex a incluir o espaço usando `\ `; ou (ii) mudando o `\mathrm{...}` pra `text{...}` (mais ou menos equivalente a `\textrm{...}`)

Reply:
Aparentemente essa parte aparecia certo no Jupyter, mas não no notebook do GitHub

Reply:
Aparentemente a definição da política na linha de cima também não aparece no github